packages/create-amplify/src/package_manager.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  NpmProjectInitializer,
+  PnpmProjectInitializer,
+  YarnClassicProjectInitializer,
+  YarnModernProjectInitializer,
+} from './project-initializer/index.js';
+import {
+  NpmPackageManagerController,
+  PnpmPackageManagerController,
+  YarnClassicPackageManagerController,
+  YarnModernPackageManagerController,
+} from './package-manager-controller/index.js';
+
+export type PackageManagerName =
+  | 'npm'
+  | 'yarn-classic'
+  | 'yarn-modern'
+  | 'pnpm';
+type PackageManagerExecutable = 'npm' | 'yarn' | 'pnpm';
+type PackageManagerBinaryRunner = 'npx' | 'yarn' | 'pnpm';
+type PackageManagerInstallCommand = 'install' | 'add';
+type PackageManagerLockFile =
+  | 'package-lock.json'
+  | 'yarn.lock'
+  | 'pnpm-lock.yaml';
+type PackageManagerInitDefault = Readonly<string[]>;
+export type PackageManager = {
+  name: PackageManagerName;
+  executable: PackageManagerExecutable;
+  binaryRunner: PackageManagerBinaryRunner;
+  installCommand: PackageManagerInstallCommand;
+  lockFile: PackageManagerLockFile;
+  initDefault: PackageManagerInitDefault;
+  projectInitializer:
+    | typeof NpmProjectInitializer
+    | typeof YarnClassicProjectInitializer
+    | typeof YarnModernProjectInitializer
+    | typeof PnpmProjectInitializer;
+  packageManagerController:
+    | typeof NpmPackageManagerController
+    | typeof YarnClassicPackageManagerController
+    | typeof YarnModernPackageManagerController
+    | typeof PnpmPackageManagerController;
+};
+export type PackageManagers = {
+  [key in PackageManagerName]: PackageManager;
+};
+export const packageManagers: PackageManagers = {
+  npm: {
+    name: 'npm',
+    executable: 'npm',
+    binaryRunner: 'npx',
+    installCommand: 'install',
+    lockFile: 'package-lock.json',
+    initDefault: ['init', '--yes'],
```

Comment:
all of these values should become internal to the PackageManagerController. Then this object can just be a map from package manager name to the corresponding projectInitializer and packageManagerController constructor

Reply:
Done